addons/sale_gelato/models/res_partner.py

```diff
@@ -9,15 +9,19 @@ class ResPartner(models.Model):
     _inherit = 'res.partner'
 
     def _gelato_prepare_address_payload(self):
+        """Trim address fields according to maximum length allowed by Gelato."""
         first_name, last_name = payment_utils.split_partner_name(self.name)
+        address_2 = self.street2 or ''
+        if remaining_address := self.street[35:]:
+            address_2 = remaining_address + ' ' + address_2
         return {
-            'companyName': self.commercial_company_name or '',
-            'firstName': first_name or last_name,  # Gelato require a first name.
-            'lastName': last_name,
-            'addressLine1': self.street,
-            'addressLine2': self.street2 or '',
+            'companyName': (self.commercial_company_name or '')[:60],
+            'firstName': (first_name or last_name)[:25],  # Gelato require a first name.
+            'lastName': last_name[:25],
+            'addressLine1': self.street[:35],
+            'addressLine2': address_2,
```

Comment:
should we truncate this as well?
```suggestion
            'addressLine2': address_2[:35],
```